src/lib/comms/sol-coap.c

```diff
@@ -1156,6 +1156,9 @@ sol_coap_packet_add_uri_path_option(struct sol_coap_packet *pkt, const char *uri
         return -EINVAL;
     }
 
+    if (strlen(uri) == 1)
+        return 0;
+
```

Comment:
This looks ok. But I would do this differently: I would move `r` to the outside scope, initialize it to zero, and instead of returning `-EINVAL` at the end, return `r`. But I am leaving this as optional, as, arguably, your version is clearer.